hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java

```diff
@@ -381,6 +383,28 @@ public static List<Path> getArchivedWALFiles(Configuration conf, ServerName serv
     return archivedWalFiles;
   }
 
+  /**
+   * List all the wal files for a logPrefix.
+   */
+  public static List<Path> getWALFiles(Configuration c, String logPrefix) throws IOException {
+    Path walRoot = new Path(CommonFSUtils.getWALRootDir(c), HConstants.HREGION_LOGDIR_NAME);
+    FileSystem fs = walRoot.getFileSystem(c);
+    List<Path> walFiles = new ArrayList<>();
+    try {
+      RemoteIterator<LocatedFileStatus> it = fs.listFiles(walRoot, true);
```

Comment:
I do not think we need recursive list here? All wals for a live regionserver will be placed under a special directory, just list the directory is enough...